content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -324,16 +326,54 @@ single `.groovy` file which interact with each other, for example:
 ----
 // vars/acme.groovy
 def setName(value) {
-    this.name = value;
+    this.name = value
 }
 def getName() {
-    return this.name;
+    return this.name
 }
 def caution(message) {
     echo "Hello, ${this.name}! CAUTION: ${message}"
 }
 ----
 
+Note that in the above, `this.name` is not referring to a field, but to an entry in a `Script.binding`, so it is recommended to use a real class definition:
+
+[source,groovy]
+----
+// vars/acme.groovy
+class acme implements Serializable {
+    private String name
+    def setName(value) {
+        this.name = value
+    }
+    def getName() {
+        return this.name
+    }
+    def caution(message) {
+        echo "Hello, ${this.name}! CAUTION: ${message}"
+    }
+}
+----
+
+or more simply:
```

Comment:
Which of these is the recommended way?  Let's use just one. 

If you want to use the second one, add callouts explaining the simplifications.
http://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code